scribus/scribusview.cpp

```diff
@@ -1658,8 +1658,13 @@ bool ScribusView::slotSetCurs(int x, int y)
 			point.setX(textFrame->width() - point.x());
 		if (textFrame->imageFlippedV())
 			point.setY(textFrame->height() - point.y());
-		textFrame->itemText.setCursorPosition(textFrame->itemText.length() == 0 ? 0 :
-			textFrame->textLayout.screenToPosition(point));
+		if (textFrame->textLayout.screenToPosition(point) >= 0)
+		{
+			if (textFrame->itemText.length() == 0)
+				textFrame->itemText.setCursorPosition(0);
+			else
+				textFrame->itemText.setCursorPosition(textFrame->textLayout.screenToPosition(point));
+		}
```

Comment:
This condition is reversed, you should check `textFrame->itemText.length() == 0` before calling `textFrame->textLayout.screenToPosition(point)`, like the old code was doing.